hnn_core/network.py

```diff
@@ -374,23 +381,31 @@ def add_evoked_drive(self, name, *, mu, sigma, numspikes,
         if not self._legacy_mode:
             _check_drive_parameter_values('evoked', sigma=sigma,
                                           numspikes=numspikes)
-
         drive = _NetworkDrive()
         drive['type'] = 'evoked'
         if name == 'extgauss':
             drive['type'] = 'gaussian'  # XXX needed to pass legacy tests!
-        drive['cell_specific'] = True
+        if n_drive_cells == 'n_cells':
```

Comment:
```suggestion
        _validate_type(n_drive_cells, (int, str), 'n_drive_cells', 'int or str')
        if n_drive_cells == 'n_cells':
            _check_option('n_drive_cells', n_drive_cells, ['n_cells']) 
```

Reply:
No need for type checks later on?

Reply:
I've put some of these checks downstream in `_attach_drive()`. Does that work or is there a good reason to put them in `add_xxx_drive()`?